### ISSUE TITLE
Bookmarks randomly move after adding a synced device

### ISSUE THREAD

**Issue by arisAlexis** [brave/brave-browser#6034]
## Description 

After adding the 4th synced device in the chain, bookmarks start to randomly move. As an example I have A,B,C,D in my bookmarks bar, *while* browsing the web they will become B,D,C,A and then I move them around and fix them and they again randomly move (which is quite frustrating and odd).

## Steps to Reproduce

   1. Add new sync device and sync it (Linux Mint 19)
   2. Use new device
   3. Switch to previous device (Windows 10) and bookmarks start moving around

## Actual result:

Bookmarks move around

## Expected result:
Bookmarks remain in their position especially during the same session/while browsing internet

## Reproduces how often: 

Easily (for my stack at least)

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Not sure
- Can you reproduce this issue with the dev channel? Not sure
- Can you reproduce this issue with the nightly channel? Not sure

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No
- Does the issue resolve itself when disabling Brave Rewards? No
- Is the issue reproducible on the latest version of Chrome?  Not sure

**Comment by pcdevil** [comment]
I can confirm the behaviour.

I recently started a new sync cycle because of this issue, but now when I added a new device (for the second cycle), it reappeared.

There are two symptoms: 
- my browsers keep getting back previously deleted bookmarks even though I _re-delete_ them when I notice it
- bookmark entries are getting duplicated

In both cases not every bookmark is affected: some which I deleted never appear again, and not all bookmarks duplicate.

I assume this is due my browsers are stuck in an intermediate state and it tries to resolve it by adding all the bookmarks, but it looks like it doesn't save the new status and later it does the same thing again.
